modules/appeals_api/app/workers/appeals_api/appeal_received_job.rb

```diff
@@ -0,0 +1,143 @@
+# @option email_identifier [String] :id_value Either the email or ICN number associated with the appellant. Required.
+# @option email_identifier [String] :id_type The type of id value provided: 'email' or 'ICN'. Required.
+# @option opts [String] :first_name First name of Veteran associated with the appeal. Required.
+# @option opts [String] :date_submitted The date of the appeal's submission. Required.
+# @option opts [String] :guid The related appeal's ID. Required.
+# @option opts [String] :claimant_email The non-Veteran claimant's email address.
+# @option opts [String] :claimant_first_name The non-Veteran claimant's first name.
+
+require 'sidekiq'
+
+module AppealsApi
+  class AppealReceivedJob
+    include Sidekiq::Worker
+
+    def perform(opts)
+      @opts = opts
+
+      return Rails.logger.error 'AppealReceived: Missing required keys' unless required_keys?
+
+      send(opts['receipt_event'].to_sym)
+    end
+
+    def hlr_received
+      return unless Flipper.enabled?(:decision_review_hlr_email)
+
+      return log_error(guid, 'HLR') unless valid_email_identifier?
+
+      template_type = 'higher_level_review_received'
+      template = { template_id: template_id(template_type) }
+
+      vanotify_service.send_email(params(template))
+    end
+
+    def nod_received
+      return unless Flipper.enabled?(:decision_review_nod_email)
+
+      return log_error(guid, 'NOD') unless valid_email_identifier?
+
+      template_type = +'notice_of_disagreement_received'
+      template = { template_id: template_id(template_type) }
+
+      vanotify_service.send_email(params(template))
+    end
+
+    def sc_received
+      return unless Flipper.enabled?(:decision_review_sc_email)
+
+      return log_error(guid, 'SC') unless valid_email_identifier?
+
+      template_type = 'supplemental_claim_received'
+      template = { template_id: template_id(template_type) }
+
+      vanotify_service.send_email(params(template))
+    end
+
+    private
+
+    attr_accessor :opts
+
+    def vanotify_service
+      @vanotify_service ||= VaNotify::Service.new(Settings.vanotify.services.lighthouse.api_key)
+    end
+
+    def params(template_opts)
+      [
+        lookup,
+        template_opts,
+        personalisation
+      ].reduce(&:merge)
+    end
+
+    def lookup
+      return { email_address: opts['claimant_email'] } if opts['claimant_email'].present?
+
+      if opts['email_identifier']['id_type'] == 'email'
+        { email_address: opts['email_identifier']['id_value'] }
+      else
+        { recipient_identifier: { id_value: opts['email_identifier']['id_value'],
+                                  id_type: opts['email_identifier']['id_type'] } }
+      end
+    end
+
+    def template_id(template)
+      t = claimant? ? "#{template}_claimant" : template
+      template_id = Settings.vanotify.services.lighthouse.template_id.public_send(t)
+      return Rails.logger.error "AppealReceived: could not find template id for #{t}" if template_id.blank?
```

Comment:
This won't return from the entire job if the template ID is blank, only from the `template_id` method, which means it'll try to send whatever "Rails.logger.error" returns as the template_id, and I'm pretty sure VANotify isn't prepared to work with that :) 

This check should be extracted to the `nod/hlr/sc_received` methods.